sidecar/internal/util/util.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// ValidateControllerEndpoint validate endpoint to make sure it contains
+// valid ip address and port number.
+func ValidateControllerEndpoint(endpoint *string) error {
+	separator := ":"
+	splits := strings.Split(*endpoint, separator)
+	if len(splits) != 2 {
+		return fmt.Errorf("invalid endpoint %q, expected format '<ip>:<port>'", *endpoint)
+	}
+
+	rawIP, rawPort := splits[0], splits[1]
+
+	ip := net.ParseIP(rawIP)
+	if ip == nil {
+		return fmt.Errorf("invalid ip address %q in endpoint %q", rawIP, *endpoint)
+	}
+
+	_, err := strconv.ParseUint(splits[1], 10, 16)
+	if err != nil {
+		return fmt.Errorf("invalid port %q in endpoint %q: %w", rawPort, *endpoint, err)
+	}
```

Comment:
possible to use https://pkg.go.dev/inet.af/netaddr#ParseIPPort?

Reply:
> possible to use https://pkg.go.dev/inet.af/netaddr#ParseIPPort?

https://github.com/inetaf/netaddr

I was not sure, since they use a BSD-3 and the project is in very early stages. 
If that's fine, then I'll change it to use this function.

Reply:
>This package is mature, optimized, and used heavily in production at Tailscale. However, API stability is not yet guaranteed.

it's not in the early stage but stability is not guaranteed. am fine with this change for now.